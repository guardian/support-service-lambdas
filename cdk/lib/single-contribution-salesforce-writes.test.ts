import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {
	APP_NAME,
	SingleContributionSalesforceWrites,
} from './single-contribution-salesforce-writes';

describe('The SingleContributionSalesforceWrites stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const codeStack = new SingleContributionSalesforceWrites(
			app,
			`${APP_NAME}-CODE`,
			{ stack: 'membership', stage: 'CODE' },
		);
		const prodStack = new SingleContributionSalesforceWrites(
			app,
			`${APP_NAME}-PROD`,
			{ stack: 'membership', stage: 'PROD' },
		);
		expect(Template.fromStack(codeStack).toJSON()).toMatchSnapshot();
		expect(Template.fromStack(prodStack).toJSON()).toMatchSnapshot();
	});
});
