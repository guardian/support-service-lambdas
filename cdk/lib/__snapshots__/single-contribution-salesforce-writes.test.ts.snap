// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SingleContributionSalesforceWrites stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuLambdaFunction",
      "GuAlarm",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AcquisitionBusToSingleContributionSalesforceWritesQueueRule2CC5A818": {
      "Properties": {
        "Description": "Send payment api events to the single-contribution-salesforce-writes-queue",
        "EventBusName": "acquisitions-bus-CODE",
        "EventPattern": {
          "account": [
            {
              "Ref": "AWS::AccountId",
            },
          ],
          "region": [
            {
              "Ref": "AWS::Region",
            },
          ],
          "source": [
            {
              "prefix": "payment-api",
            },
          ],
        },
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::GetAtt": [
                "singlecontributionsalesforcewritesqueue227D3F6D",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "deadletterssinglecontributionsalesforcewritesqueueA3CB7F4A": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "dead-letters-single-contribution-salesforce-writes-queue-CODE",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "singlecontributionsalesforcewritesalarm02E37F08": {
      "Properties": {
        "ActionsEnabled": false,
        "AlarmActions": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:sns:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":",
                {
                  "Fn::GetAtt": [
                    "singlecontributionsalesforcewritestopicD1E1FC97",
                    "TopicName",
                  ],
                },
              ],
            ],
          },
        ],
        "AlarmDescription": {
          "Fn::Join": [
            "",
            [
              "Impact: customer service representative cannot see single contribution in Salesforce. Fix: check logs for lambda ",
              {
                "Ref": "singlecontributionsalesforcewriteslambda1B9830FE",
              },
            ],
          ],
        },
        "AlarmName": "CODE: Failed to sync single contribution to Salesforce",
        "ComparisonOperator": "GreaterThanThreshold",
        "Dimensions": [
          {
            "Name": "QueueName",
            "Value": {
              "Fn::GetAtt": [
                "deadletterssinglecontributionsalesforcewritesqueueA3CB7F4A",
                "QueueName",
              ],
            },
          },
        ],
        "EvaluationPeriods": 24,
        "MetricName": "ApproximateNumberOfMessagesVisible",
        "Namespace": "AWS/SQS",
        "Period": 3600,
        "Statistic": "Sum",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "singlecontributionsalesforcewriteslambda1B9830FE": {
      "DependsOn": [
        "singlecontributionsalesforcewriteslambdaServiceRoleDefaultPolicy67F57101",
        "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "membership/CODE/single-contribution-salesforce-writes/single-contribution-salesforce-writes.jar",
        },
        "Environment": {
          "Variables": {
            "APP": "single-contribution-salesforce-writes",
            "STACK": "membership",
            "STAGE": "CODE",
          },
        },
        "FunctionName": "single-contribution-salesforce-writes-CODE",
        "Handler": "com.gu.singleContributionSalesforceWrites.handlers.CreateSalesforceSingleContributionRecordHandler::handleRequest",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A",
            "Arn",
          ],
        },
        "Runtime": "java11",
        "Tags": [
          {
            "Key": "App",
            "Value": "single-contribution-salesforce-writes",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "single-contribution-salesforce-writes",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "singlecontributionsalesforcewriteslambdaServiceRoleDefaultPolicy67F57101": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "singlecontributionsalesforcewritesqueue227D3F6D",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/membership/CODE/single-contribution-salesforce-writes/single-contribution-salesforce-writes.jar",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/membership/single-contribution-salesforce-writes",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/membership/single-contribution-salesforce-writes/*",
                  ],
                ],
              },
            },
            {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:secretsmanager:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":secret:CODE/Salesforce/User/SingleContributionSalesforceWrites-*",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:secretsmanager:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":secret:CODE/Salesforce/ConnectedApp/SingleContributionSalesforceWrites-*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "singlecontributionsalesforcewriteslambdaServiceRoleDefaultPolicy67F57101",
        "Roles": [
          {
            "Ref": "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "singlecontributionsalesforcewriteslambdaSqsEventSourcesinglecontributionsalesforcewritesCODEsinglecontributionsalesforcewritesqueue3444812FD893DC08": {
      "Properties": {
        "EventSourceArn": {
          "Fn::GetAtt": [
            "singlecontributionsalesforcewritesqueue227D3F6D",
            "Arn",
          ],
        },
        "FunctionName": {
          "Ref": "singlecontributionsalesforcewriteslambda1B9830FE",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "singlecontributionsalesforcewritesqueue227D3F6D": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "single-contribution-salesforce-writes-queue-CODE",
        "RedrivePolicy": {
          "deadLetterTargetArn": {
            "Fn::GetAtt": [
              "deadletterssinglecontributionsalesforcewritesqueueA3CB7F4A",
              "Arn",
            ],
          },
          "maxReceiveCount": 3,
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "singlecontributionsalesforcewritesqueuePolicyDC24ED87": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:SendMessage",
                "sqs:GetQueueAttributes",
                "sqs:GetQueueUrl",
              ],
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Fn::GetAtt": [
                      "AcquisitionBusToSingleContributionSalesforceWritesQueueRule2CC5A818",
                      "Arn",
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "singlecontributionsalesforcewritesqueue227D3F6D",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sqs:SendMessage",
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Fn::GetAtt": [
                      "AcquisitionBusToSingleContributionSalesforceWritesQueueRule2CC5A818",
                      "Arn",
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "singlecontributionsalesforcewritesqueue227D3F6D",
                  "Arn",
                ],
              },
              "Sid": "Allow acquisition bus to send messages to the single-contribution-salesforce-writes-queue",
            },
          ],
          "Version": "2012-10-17",
        },
        "Queues": [
          {
            "Ref": "singlecontributionsalesforcewritesqueue227D3F6D",
          },
        ],
      },
      "Type": "AWS::SQS::QueuePolicy",
    },
    "singlecontributionsalesforcewritestopicD1E1FC97": {
      "Properties": {
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "TopicName": "single-contribution-salesforce-writes-topic-CODE",
      },
      "Type": "AWS::SNS::Topic",
    },
    "singlecontributionsalesforcewritestopicsupporterrevenueengineguardiancouk88BB94BA": {
      "Properties": {
        "Endpoint": "supporter.revenue.engine@guardian.co.uk",
        "Protocol": "email",
        "TopicArn": {
          "Ref": "singlecontributionsalesforcewritestopicD1E1FC97",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`The SingleContributionSalesforceWrites stack matches the snapshot 2`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuLambdaFunction",
      "GuAlarm",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AcquisitionBusToSingleContributionSalesforceWritesQueueRule2CC5A818": {
      "Properties": {
        "Description": "Send payment api events to the single-contribution-salesforce-writes-queue",
        "EventBusName": "acquisitions-bus-PROD",
        "EventPattern": {
          "account": [
            {
              "Ref": "AWS::AccountId",
            },
          ],
          "region": [
            {
              "Ref": "AWS::Region",
            },
          ],
          "source": [
            {
              "prefix": "payment-api",
            },
          ],
        },
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::GetAtt": [
                "singlecontributionsalesforcewritesqueue227D3F6D",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "deadletterssinglecontributionsalesforcewritesqueueA3CB7F4A": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "dead-letters-single-contribution-salesforce-writes-queue-PROD",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "singlecontributionsalesforcewritesalarm02E37F08": {
      "Properties": {
        "ActionsEnabled": true,
        "AlarmActions": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:sns:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":",
                {
                  "Fn::GetAtt": [
                    "singlecontributionsalesforcewritestopicD1E1FC97",
                    "TopicName",
                  ],
                },
              ],
            ],
          },
        ],
        "AlarmDescription": {
          "Fn::Join": [
            "",
            [
              "Impact: customer service representative cannot see single contribution in Salesforce. Fix: check logs for lambda ",
              {
                "Ref": "singlecontributionsalesforcewriteslambda1B9830FE",
              },
            ],
          ],
        },
        "AlarmName": "PROD: Failed to sync single contribution to Salesforce",
        "ComparisonOperator": "GreaterThanThreshold",
        "Dimensions": [
          {
            "Name": "QueueName",
            "Value": {
              "Fn::GetAtt": [
                "deadletterssinglecontributionsalesforcewritesqueueA3CB7F4A",
                "QueueName",
              ],
            },
          },
        ],
        "EvaluationPeriods": 24,
        "MetricName": "ApproximateNumberOfMessagesVisible",
        "Namespace": "AWS/SQS",
        "Period": 3600,
        "Statistic": "Sum",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "singlecontributionsalesforcewriteslambda1B9830FE": {
      "DependsOn": [
        "singlecontributionsalesforcewriteslambdaServiceRoleDefaultPolicy67F57101",
        "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "membership/PROD/single-contribution-salesforce-writes/single-contribution-salesforce-writes.jar",
        },
        "Environment": {
          "Variables": {
            "APP": "single-contribution-salesforce-writes",
            "STACK": "membership",
            "STAGE": "PROD",
          },
        },
        "FunctionName": "single-contribution-salesforce-writes-PROD",
        "Handler": "com.gu.singleContributionSalesforceWrites.handlers.CreateSalesforceSingleContributionRecordHandler::handleRequest",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A",
            "Arn",
          ],
        },
        "Runtime": "java11",
        "Tags": [
          {
            "Key": "App",
            "Value": "single-contribution-salesforce-writes",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "single-contribution-salesforce-writes",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "singlecontributionsalesforcewriteslambdaServiceRoleDefaultPolicy67F57101": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "singlecontributionsalesforcewritesqueue227D3F6D",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/membership/PROD/single-contribution-salesforce-writes/single-contribution-salesforce-writes.jar",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/membership/single-contribution-salesforce-writes",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/membership/single-contribution-salesforce-writes/*",
                  ],
                ],
              },
            },
            {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:secretsmanager:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":secret:PROD/Salesforce/User/SingleContributionSalesforceWrites-*",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:secretsmanager:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":secret:PROD/Salesforce/ConnectedApp/SingleContributionSalesforceWrites-*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "singlecontributionsalesforcewriteslambdaServiceRoleDefaultPolicy67F57101",
        "Roles": [
          {
            "Ref": "singlecontributionsalesforcewriteslambdaServiceRole4866AA0A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "singlecontributionsalesforcewriteslambdaSqsEventSourcesinglecontributionsalesforcewritesPRODsinglecontributionsalesforcewritesqueueD0DDCB72DC4C762C": {
      "Properties": {
        "EventSourceArn": {
          "Fn::GetAtt": [
            "singlecontributionsalesforcewritesqueue227D3F6D",
            "Arn",
          ],
        },
        "FunctionName": {
          "Ref": "singlecontributionsalesforcewriteslambda1B9830FE",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "singlecontributionsalesforcewritesqueue227D3F6D": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "single-contribution-salesforce-writes-queue-PROD",
        "RedrivePolicy": {
          "deadLetterTargetArn": {
            "Fn::GetAtt": [
              "deadletterssinglecontributionsalesforcewritesqueueA3CB7F4A",
              "Arn",
            ],
          },
          "maxReceiveCount": 3,
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "singlecontributionsalesforcewritesqueuePolicyDC24ED87": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:SendMessage",
                "sqs:GetQueueAttributes",
                "sqs:GetQueueUrl",
              ],
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Fn::GetAtt": [
                      "AcquisitionBusToSingleContributionSalesforceWritesQueueRule2CC5A818",
                      "Arn",
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "singlecontributionsalesforcewritesqueue227D3F6D",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sqs:SendMessage",
              "Condition": {
                "ArnEquals": {
                  "aws:SourceArn": {
                    "Fn::GetAtt": [
                      "AcquisitionBusToSingleContributionSalesforceWritesQueueRule2CC5A818",
                      "Arn",
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
              "Resource": {
                "Fn::GetAtt": [
                  "singlecontributionsalesforcewritesqueue227D3F6D",
                  "Arn",
                ],
              },
              "Sid": "Allow acquisition bus to send messages to the single-contribution-salesforce-writes-queue",
            },
          ],
          "Version": "2012-10-17",
        },
        "Queues": [
          {
            "Ref": "singlecontributionsalesforcewritesqueue227D3F6D",
          },
        ],
      },
      "Type": "AWS::SQS::QueuePolicy",
    },
    "singlecontributionsalesforcewritestopicD1E1FC97": {
      "Properties": {
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "TopicName": "single-contribution-salesforce-writes-topic-PROD",
      },
      "Type": "AWS::SNS::Topic",
    },
    "singlecontributionsalesforcewritestopicsupporterrevenueengineguardiancouk88BB94BA": {
      "Properties": {
        "Endpoint": "supporter.revenue.engine@guardian.co.uk",
        "Protocol": "email",
        "TopicArn": {
          "Ref": "singlecontributionsalesforcewritestopicD1E1FC97",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;
